src/UbiTcp.cpp

```diff
@@ -0,0 +1,223 @@
+
+}
+
+float UbiTcp::get(const char* device_label, const char* variable_label) {
+  /* Connecting the client */
+  _client_tcp_ubi.connect(_host, UBIDOTS_TCP_PORT);
+  reconnect(_host, UBIDOTS_TCP_PORT);
+
+  if (_client_tcp_ubi.connected()) {
+    /* Builds the request POST - Please reference this link to know all the request's structures https://ubidots.com/docs/api/ */
+    _client_tcp_ubi.print(_user_agent);
+    _client_tcp_ubi.print("|LV|");
+    _client_tcp_ubi.print(_token);
+    _client_tcp_ubi.print("|");
+    _client_tcp_ubi.print(device_label);
+    _client_tcp_ubi.print(":");
+    _client_tcp_ubi.print(variable_label);
+    _client_tcp_ubi.print("|end");
+
+    if (_debug){
+      Serial.println("----");
+      Serial.println("Payload for request:");
+      Serial.print(_user_agent);
+      Serial.print("|LV|");
+      Serial.print(_token);
+      Serial.print("|");
+      Serial.print(device_label);
+      Serial.print(":");
+      Serial.print(variable_label);
+      Serial.print("|end");
+      Serial.println("\n----");
+    }
+
+    /* Waits for the host's answer */
+    if (!waitServerAnswer()) {
+      _client_tcp_ubi.stop();
+      return ERROR_VALUE;
+    }
+
+    /* Reads the response from the server */
+    char* response = (char *) malloc(sizeof(char) * MAX_BUFFER_SIZE);
+    float value = parseTcpAnswer("LV", response);
+    _client_tcp_ubi.stop();
+    free(response);
+    return value;
+  }
+
+  if (_debug) {
+    Serial.println("ERROR could not connect to the server");
+  }
+
+  _client_tcp_ubi.stop();
+  return ERROR_VALUE;
+}
+
+/**
+ * Reconnects to the server
+ * @return true once the host answer buffer length is greater than zero,
+ *         false if timeout is reached.
+ */
+
+void UbiTcp::reconnect(const char * host, const int port) {
+  uint8_t attempts = 0;
+  while (!_client_tcp_ubi.connected() && attempts < 5) {
+    if (_debug) {
+      Serial.print("Trying to connect to ");
+      Serial.print(host);
+      Serial.print(" , attempt number: ");
+      Serial.println(attempts);
+    }
+    _client_tcp_ubi.connect(host, port);
+    attempts += 1;
+    delay(1000);
+  }
+}
+
+/**
+ * Function to wait for the host answer up to the already set _timeout.
+ * @return true once the host answer buffer length is greater than zero,
+ *         false if timeout is reached.
+ */
+
+bool UbiTcp::waitServerAnswer() {
+  int timeout = 0;
+  while(!_client_tcp_ubi.available() && timeout < _timeout) {
+    timeout++;
+    delay(1);
+    if (timeout > _timeout - 1) {
+      if (_debug) {
+        Serial.println("timeout, could not read any response from the host");
+      }
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ * Parse the TCP host answer and saves it to the input char pointer.
+ * @payload [Mandatory] char payload pointer to store the host answer.
+ * @return true if there is an 'Ok' in the answer, false if not.
+ */
+
+float UbiTcp::parseTcpAnswer(const char* request_type, char* response) {
```

Comment:
change to parseTCPAnswer

Reply:
The standard does not fit with that, https://google.github.io/styleguide/cppguide.html#General_Naming_Rules